src/eviction/lru.c

```diff
@@ -224,49 +224,88 @@ static void balance_lru_list(ocf_cache_t cache,
 
 /*-- End of LRU functions*/
 
-void evp_lru_init_cline(ocf_cache_t cache, ocf_cache_line_t cline)
-{
-	struct lru_eviction_policy_meta *node;
-
-	node = &ocf_metadata_get_eviction_policy(cache, cline)->lru;
-
-	node->hot = false;
-	node->prev = end_marker;
-	node->next = end_marker;
-}
-
-static struct ocf_lru_list *evp_lru_get_list(struct ocf_user_part *part,
+static struct ocf_lru_list *evp_lru_get_list(struct ocf_part_runtime *part,
 		uint32_t evp, bool clean)
 {
-	return clean ? &part->runtime->eviction[evp].policy.lru.clean :
-			&part->runtime->eviction[evp].policy.lru.dirty;
+	return clean ? &part->eviction[evp].policy.lru.clean :
+			&part->eviction[evp].policy.lru.dirty;
 }
 
 static inline struct ocf_lru_list *evp_get_cline_list(ocf_cache_t cache,
 		ocf_cache_line_t cline)
 {
-	ocf_part_id_t part_id = ocf_metadata_get_partition_id(cache, cline);
-	struct ocf_user_part *part = &cache->user_parts[part_id];
 	uint32_t ev_list = (cline % OCF_NUM_EVICTION_LISTS);
+	ocf_part_id_t part_id;
+	struct ocf_part_runtime *part;
+
+	part_id = ocf_metadata_get_partition_id(cache, cline);
+	ENV_BUG_ON(part_id == PARTITION_UNSPECIFIED ||
+			part_id == PARTITION_INVALID);
```

Comment:
`ENV_BUG_ON(part_id > OCF_IO_CLASS_ID_MAX);` would do the trick.

Reply:
after adding ocf_part struct this is changed to "part_id == PARTITION_UNSPECIFIED" as PARTITION_FREELIST is now a valid input to this function